src/ImageSmooth.jl

```diff
@@ -20,6 +20,9 @@ export
     # generic API
     smooth, smooth!,
 
+    # forward difference and backward difference
+    forwarddiff, forwarddiff!, backdiff, backdiff!,
+
```

Comment:
I don't think we should export these utils, especially since we plan to move them to `ImageBase`.

Reply:
I've proposed `fdiff` in https://github.com/JuliaImages/ImageBase.jl/pull/11, when that's merged, then we can use that version here.

Because this PR is quite large, we can do this in the next PR and leave the current `forwarddiff`/`backdiff` as they are. No need to change since we eventually will remove them.